chore: set BIP-44 as default

gantunesr opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Force the multichain accounts feature flag to be true by default, as it is "fuse" (cannot be reversed)

Most of this PR is about allowing the tests to continue as if the legacy state was the case, while we update them.

For unit tests, this involves mocking the feature flag wrapper. For E2E tests we do the same thing but by mocking the feature flag API (which can be disabled in a number of instances)

In some instances, this involved changing the metametrics ID used in fixtures, because it interferes with the mock functionality (due to segmentation happening closer to the core)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36254?quickstart=1)

## **Changelog**

CHANGELOG entry: set BIP-44/multichain accounts as the default regardless of the remote feature flag

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/36678
Fixes: https://consensyssoftware.atlassian.net/jira/software/c/projects/MUL/boards/1258/backlog?assignee=5cfa627c9943230e77ad45eb&selectedIssue=MUL-1089

## **Manual testing steps**

Not user facing

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.



<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Default-enables multichain accounts (BIP-44) and updates tests, mocks, helpers, and snapshots to support legacy/testing states and new defaults.
> 
> - **Shared/Core**:
>   - Update `shared/lib/multichain-accounts/remote-feature-flag.ts` to default feature enabled when flag is undefined/null/empty; tighten enablement logic and types. Adjust tests to expect default `true`.
> - **Test Infrastructure (E2E)**:
>   - Add `forceBip44Version` option in `test/e2e/helpers.js` to choose BIP-44 state (0/1/2) with corresponding mocks from new `tests/multichain-accounts/feature-flag-mocks`.
>   - Apply `forceBip44Version: 2` in multiple E2E suites; remove inline multichain flag mocks where replaced; skip some multichain/BTC suites.
>   - Enhance WebDriver `waitForMultipleSelectors` to accept `{ timeout, state }` and fix internal waits.
>   - Add QR code interactions in `AccountListPage` and `MultichainAccountDetailsPage`; update selectors.
> - **UI/Unit/Integration Tests**:
>   - Mock multichain selectors (`getIsMultichainAccountsState1/2Enabled`) to `false` in numerous unit tests to preserve legacy behavior; update snapshots (e.g., network row rendering, font-weight `500`, header changes).
>   - Set default remote feature flags in test render helpers and integration init state.
> - **Metrics/Other Tests**:
>   - Use `MOCK_META_METRICS_ID` and `loginWithoutBalanceValidation` where needed; adjust timings and state snapshots.
> - **Misc**:
>   - Add centralized multichain feature-flag mocks module; update various tests to use it; minor log and fixture tweaks.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fdf1ac7afea63886c832e4b5ebf60cab0c6001d5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

gantunesr committed:
chore: set BIP-44 as default

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-extension into gar/chore/set-bip44-default

ngalee8311 commented:
gar/chore/set-bip44-default
files changed

shane-t committed:
chore: update snapshots

shane-t commented:
FAIL  ui/pages/confirmations/components/confirm/header/header.test.tsx (51.147 s)
 FAIL  ui/pages/confirmations/confirm/confirm.test.tsx (62.186 s)
 FAIL  ui/components/multichain/pages/send/components/account-picker.test.tsx (21.03 s)
 FAIL  ui/components/multichain/pages/send/send.test.js (18.75 s)
 FAIL  ui/components/multichain/app-header/app-header.test.js (19.093 s)
 FAIL  app/scripts/lib/util.test.js (5.076 s)
 FAIL  ui/components/multichain/account-picker/account-picker.test.js (27.546 s)
 FAIL  ui/components/multichain-accounts/add-wallet-modal/add-wallet-modal.test.tsx (9.78 s)
 FAIL  app/scripts/metamask-controller.test.js (51.266 s)
 FAIL  ui/components/multichain/network-list-menu/network-list-menu.test.tsx (71.504 s)
 FAIL  ui/pages/onboarding-flow/import-srp/import-srp.test.js (50.589 s)
 FAIL  ui/pages/routes/routes.component.test.js (27.735 s)
 FAIL  app/scripts/controllers/account-tracker-controller.test.ts (222.207 s)
 FAIL  app/scripts/metamask-controller.actions.test.js (36.437 s)
 FAIL  ui/pages/confirmations/components/confirm/info/info.test.tsx (29.059 s)
 FAIL  ui/components/multichain/account-details/account-details.test.js (46.026 s)
 FAIL  ui/pages/confirmations/components/confirm/header/header.test.tsx (42.171 s)
 FAIL  ui/components/multichain/app-header/app-header.test.js (45.28 s)
 FAIL  app/scripts/lib/util.test.js (20.081 s)
 FAIL  ui/components/multichain-accounts/add-wallet-modal/add-wallet-modal.test.tsx (47.516 s)
 FAIL  app/scripts/metamask-controller.test.js (80.57 s)
 FAIL  ui/components/multichain/network-list-menu/network-list-menu.test.tsx (139.704 s)
 FAIL  ui/components/multichain/edit-accounts-modal/edit-accounts-modal.test.tsx (86.474 s)
 FAIL  ui/pages/onboarding-flow/import-srp/import-srp.test.js (28.45 s)
 FAIL  ui/pages/routes/routes.component.test.js (79.913 s)
 FAIL  app/scripts/controllers/account-tracker-controller.test.ts (359.533 s)
 FAIL  ui/components/multichain/import-tokens-modal/import-tokens-modal.test.js (84.36 s)
 FAIL  app/scripts/metamask-controller.actions.test.js (164.061 s)
 FAIL  ui/components/app/srp-input/srp-input.test.js (638.466 s)
 FAIL  ui/pages/asset/components/asset-page.test.tsx (20.099 s)
 FAIL  ui/components/multichain/account-list-menu/account-list-menu.test.tsx (23.559 s)
 FAIL  ui/components/multichain-accounts/multichain-account-list-menu/multichain-account-list-menu.test.tsx (30.532 s)
 FAIL  ui/components/multichain/account-menu/account-menu.test.tsx (12.312 s)
 FAIL  ui/components/app/wallet-overview/eth-overview.test.js (15.186 s)
 FAIL  ui/components/multichain/account-overview/account-overview-unknown.test.tsx (9.617 s)
 FAIL  ui/components/app/snaps/snap-ui-renderer/components/address-input.test.ts (26.621 s)
 FAIL  ui/components/multichain/account-list-item-menu/account-list-item-menu.test.js (14.609 s)
 FAIL  ui/components/app/wallet-overview/non-evm-overview.test.tsx (5.245 s)
 FAIL  ui/selectors/multichain-accounts/feature-flags.test.ts
 FAIL  ui/pages/confirmations/components/confirm/header/header.test.tsx (42.171 s)
 FAIL  ui/components/multichain/app-header/app-header.test.js (45.28 s)
 FAIL  app/scripts/lib/util.test.js (20.081 s)
 FAIL  ui/components/multichain-accounts/add-wallet-modal/add-wallet-modal.test.tsx (47.516 s)
 FAIL  app/scripts/metamask-controller.test.js (80.57 s)
 FAIL  ui/components/multichain/network-list-menu/network-list-menu.test.tsx (139.704 s)
 FAIL  ui/components/multichain/edit-accounts-modal/edit-accounts-modal.test.tsx (86.474 s)
 FAIL  ui/pages/onboarding-flow/import-srp/import-srp.test.js (28.45 s)
 FAIL  ui/pages/routes/routes.component.test.js (79.913 s)
 FAIL  app/scripts/controllers/account-tracker-controller.test.ts (359.533 s)
 FAIL  ui/components/multichain/import-tokens-modal/import-tokens-modal.test.js (84.36 s)
 FAIL  app/scripts/metamask-controller.actions.test.js (164.061 s)
 FAIL  ui/components/app/srp-input/srp-input.test.js (638.466 s)
 FAIL  ui/pages/asset/components/asset-page.test.tsx (20.099 s)
 FAIL  ui/components/multichain/account-list-menu/account-list-menu.test.tsx (23.559 s)
 FAIL  ui/components/multichain-accounts/multichain-account-list-menu/multichain-account-list-menu.test.tsx (30.532 s)
 FAIL  ui/components/multichain/account-menu/account-menu.test.tsx (12.312 s)
 FAIL  ui/components/app/wallet-overview/eth-overview.test.js (15.186 s)
 FAIL  ui/components/multichain/account-overview/account-overview-unknown.test.tsx (9.617 s)
 FAIL  ui/components/app/snaps/snap-ui-renderer/components/address-input.test.ts (26.621 s)
 FAIL  ui/components/multichain/account-list-item-menu/account-list-item-menu.test.js (14.609 s)
 FAIL  ui/components/app/wallet-overview/non-evm-overview.test.tsx (5.245 s)
 FAIL  ui/selectors/multichain-accounts/feature-flags.test.ts

shane-t committed:
fix: remove file

shane-t committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
chore: hot wire a few tests

shane-t committed:
chore: fix more unit tests

shane-t committed:
fix: fix more unit tests

ngalee8311 commented:
gar/chore/set-bip44-default
.patch

shane-t committed:
chore: for now, hack balance check to 0 USD

shane-t committed:
chore: more e2es

shane-t committed:
test: try a simpler (?) solution to fix the e2es

shane-t committed:
chore: revert a bunch of changes

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
chore: revert a bunch of changes

shane-t committed:
chore: update snapshots

shane-t committed:
chore: e2e hack

shane-t committed:
chore: slightly better hack

shane-t committed:
chore: refactor

shane-t committed:
chore: both flags..

shane-t committed:
chore: fix some more e2e

davibroc committed:
Disable multichain tests

davibroc committed:
Add back the mock

davibroc committed:
Remove one more tests

davibroc committed:
Fix lint

davibroc committed:
Fixed lint

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
chore: fix some tests that can be fixed with overrides

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
chore: fix lint

shane-t committed:
chore: use mock metametrics ID

shane-t committed:
chore: fix some more E2Es

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

racitores committed:
fix: integration tests

shane-t committed:
fix: more tests

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
chore: fix some more E2Es

shane-t committed:
chore: fix some more E2Es

shane-t committed:
chore: fix some more E2Es

davibroc committed:
Fixed error.spec.ts

shane-t committed:
chore: fix lint

shane-t committed:
Merge branch 'test/fix-bip44-failures' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
chore: prettier

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
chore: fix wallet_invokeMethod test

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

davibroc committed:
Fix vault e2e test (#36545)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36545?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates vault-corruption E2E to use multichain address QR flow with
Bip44 v2, adding QR helpers and adjusting selectors/timeouts.
> 
> - **E2E Tests**:
> - **Vault Corruption Flow
(`test/e2e/tests/vault-corruption/vault-corruption.spec.ts`)**:
> - Use `MultichainAccountDetailsPage` and navigate via `Addresses` → QR
to fetch first address.
>     - Set `forceBip44Version: 2` in test config.
> - **Account List Page
(`test/e2e/page-objects/pages/account-list-page.ts`)**:
> - Add `showQRCodeButton` selector and `clickShowQR(options?)` helper.
> - Increase `waitForMultipleSelectors` timeout to `30000` in
`checkPageIsLoaded`.
> - **Multichain Account Details Page
(`test/e2e/page-objects/pages/multichain-account-details-page.ts`)**:
> - Update share modal address selector to `.max-w-64` in
`getAddressFromShareModal`.
>     - Add `closeQRCodeDialog()` helper.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
76f0688f4e80c8bab1e54632ce295bbe35357980. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

gantunesr committed:
Merge branch 'main' into gar/chore/set-bip44-default

davibroc committed:
test: Fix vault test flakiness (#36575)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36575?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Refactors E2E wait utilities to accept options and properly pass
timeouts, and updates page object to use the new API.
> 
> - **E2E WebDriver**:
> - `waitForSelector`: passes `timeout` into internal `findElement` for
`detached`, `enabled`, and `disabled` states.
> - `waitForMultipleSelectors`: changes signature to accept `{ timeout,
state }` options and forwards them to `waitForSelector`; updates JSDoc
accordingly.
> - **Page Object**:
> - `test/e2e/page-objects/pages/account-list-page.ts`
`checkPageIsLoaded`: uses `waitForMultipleSelectors` with `{ timeout:
30000, state: 'visible' }`.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
6e641047b68c06b3c3c7099bbd0142d060b0bf4f. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

OGPoyraz reviewed (dismissed):
Snapshot changes are expected in confirmations as we are defaulting to BIP44 flag in this PR. (Network rows will be included because we are not showing network badge in the header with BIP44 anymore)

LGTM

fabiobozzo commented on shared/lib/multichain-accounts/remote-feature-flag.ts:
```diff
@@ -34,13 +34,14 @@ export const isMultichainAccountsFeatureEnabled = (
+    enabled &&
+    minimumVersion !== null &&
+    currentFeatureVersion === featureVersion &&
+    semver.gte(APP_VERSION, minimumVersion)
```
```suggestion
    semver.valid(minimumVersion) &&
    semver.gte(APP_VERSION, minimumVersion)
```

fabiobozzo commented on shared/lib/multichain-accounts/remote-feature-flag.test.ts:
```diff
@@ -37,13 +37,13 @@ describe('remoteFeatureFlag', () => {
   it('should return false when the feature flag is undefined', () => {
     const featureFlag = undefined;
 
-    expect(isMultichainAccountsFeatureEnabled(featureFlag, '2')).toBe(false);
+    expect(isMultichainAccountsFeatureEnabled(featureFlag, '2')).toBe(true);
```
Nit: rename these two tests to 

- "should return **true** when the feature flag is undefined"  
- "should return **true** when the feature flag is null"

shane-t committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
Update shared/lib/multichain-accounts/remote-feature-flag.ts

Co-authored-by: Fabio Bozzo <fabio.bozzo@consensys.net>

shane-t committed:
chore: test name fix

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'gar/chore/set-bip44-default' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into gar/chore/set-bip44-default

davibroc committed:
Fix mock conflict (#36690)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36690?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds `sendRedesign` (disabled) to remote feature flags in the state
snapshot and multichain feature flag mocks.
> 
> - **Tests**:
>   - **Feature Flags**:
> - Add `sendRedesign: { enabled: false }` to
`RemoteFeatureFlagController.remoteFeatureFlags` in
`test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-background-state.json`.
> - Include `sendRedesign: { enabled: false }` in
`test/e2e/tests/multichain-accounts/feature-flag-mocks.ts` disabled mock
response.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
a955c9e99b4fdf8ac0663be8916141756d13e576. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

gantunesr committed:
fix: lint

gantunesr committed:
Merge branch 'main' into gar/chore/set-bip44-default

shane-t committed:
Merge branch 'main' into gar/chore/set-bip44-default

mathieuartu committed:
fix: lint errors & test delays
